modules/terrain/src/lib/parse-terrain.js

```diff
@@ -19,27 +19,27 @@ function getTerrain(imageData, tileSize, elevationDecoder) {
     }
   }
   // backfill bottom border
-  for (let i = gridSize * (gridSize - 1), x = 0; x < gridSize - 1; x++, i++) {
-    terrain[i] = terrain[i - gridSize];
+  for (let i = width * (width - 1), x = 0; x < width - 1; x++, i++) {
+    terrain[i] = terrain[i - width];
   }
   // backfill right border
-  for (let i = gridSize - 1, y = 0; y < gridSize; y++, i += gridSize) {
+  for (let i = height - 1, y = 0; y < height; y++, i += height) {
     terrain[i] = terrain[i - 1];
   }
```

Comment:
Note that you don't need to backfill for the Delatin algorithm. Martini's algorithm uses a grid of `n + 1, n + 1`, so you need 257 x 257 pixels. With delatin there's no need to do this, and you also don't need to instantiate a `terrain` array of `(width + 1) * (height + 1)`, you can instantiate an array of just `width + height`.